### PR TITLE
Add support for SXML indentation on output.

### PR DESCRIPTION
**Pull request by arthurgleckler**
This adds a second optional argument to `sxml-display-as-html`.  If true, it adds newlines and indentation where appropriate.  This makes reading the HTML easier.

### CODE REVIEW

**Comment by lassik** [comment]
We should probably use keyword arguments or a settings object of some kind. It's not easy to predict what options will be added in the future.

File: lib/chibi/sxml.scm [hunk 1]
```diff
@@ -80,13 +80,24 @@
...
 ;;> \var{@raw} tag is considered safe text and not processed or escaped.
 (define (sxml-display-as-html sxml . o)
-  (let ((out (if (pair? o) (car o) (current-output-port))))
+  (let-optionals o ((out (current-output-port))
+		    (indent? #false))
     (let lp ((sxml (if (and (pair? sxml) (eq? '*TOP* (car sxml)))
                        (cdr sxml)
-                       sxml)))
+                       sxml))
+	     (i 0))
```
Comment (ashinn) [inline lib/chibi/sxml.scm:100]: nit: i sounds like a normal counter, maybe better to name this depth or indent?

**Review by ashinn — APPROVED** [review]
Two optional arguments is fine, no need to use keywords here.

File: lib/chibi/sxml.scm (hunk 1 above)
Comment (arthurgleckler) [reply lib/chibi/sxml.scm]: Good idea.  I renamed it `depth`.

**Pull request merged**